app/pages/learn/01_tutorial/01_your-first-java-app/02_building-with-intellij-idea.md

```diff
@@ -0,0 +1,221 @@
+
+[![Project view](/assets/images/intellij-idea/project-view.png)](/assets/images/intellij-idea/project-view.png)
+
+Let’s add some code. We’ll start by creating a new class. In the **Project** tool window on the left, select the directory `src/main/java`. Add a new Java file using the shortcut **⌘N** (on macOS) or **Alt+Insert** (on Windows/Linux). Alternatively, right-click the **Project** tool window to open the context menu and select **New | Java class**. Name this class `Main`.
+
+## Writing and editing code
+Inside the class `Main`, we will write the main method, which is where execution of our Java program will start.
+The classic way to write a main method in Java is:
+```shell
+public static void main(String[] args) {
+  
+}
+```
+As you start typing, you’ll notice that IntelliJ IDEA gives you [code completion](https://www.jetbrains.com/help/idea/auto-completing-code.html). It will help you complete the names of classes, methods, fields, and keywords, and other types of completion.
+
+[![Code completion](/assets/images/intellij-idea/code-completion1.png)](/assets/images/intellij-idea/code-completion1.png)
+
+[![Code completion](/assets/images/intellij-idea/code-completion2.png)](/assets/images/intellij-idea/code-completion2.png)
+
+IntelliJ IDEA will show you if you’ve typed or selected something that doesn’t compile, or if it sees any other problems. If you press **Alt+Enter** it will offer options to fix the problem. You can use **F2** to move to the next problem, and **Shift+F2** to go to the previous problem. IntelliJ IDEA will help you keep your code green, by offering suggestions that are context-sensitive.
+
+[![Suggestions](/assets/images/intellij-idea/alt-enter.png)](/assets/images/intellij-idea/alt-enter.png)
+
+To speed up development, we can also [generate code](https://www.jetbrains.com/help/idea/generating-code.html). IntelliJ IDEA can generate constructors, getters and setters, toString(), equals() and hashCode() methods, and more.
+
+[![Generate code](/assets/images/intellij-idea/generate-code.png)](/assets/images/intellij-idea/generate-code.png)
+
+Instead of writing the main method ourselves, we can use [live templates](https://www.jetbrains.com/help/idea/using-live-templates.html) to write it for us. When we write `main`, IntelliJ IDEA will suggest the main live template. Press **Enter** to select the suggestion, and the main method will appear.
+
+[![Main live template](/assets/images/intellij-idea/main-live-template.png)](/assets/images/intellij-idea/main-live-template.png)
+
+Now we can write the desired code inside the main method. We can use another live template, `sout`, to generate `System.out.println();` and add the String `“Hello World!”`
+
+[![Hello World](/assets/images/intellij-idea/hello-world.png)](/assets/images/intellij-idea/hello-world.png)
+
+There are several predefined live templates, and you can also [create your own](https://www.jetbrains.com/help/idea/creating-and-editing-live-templates.html).
```

Comment:
Does it make sense for beginners to know about creating their own templates?

Reply:
Removed the part about creating your own.